ts/private/ts_project_worker.js

```diff
@@ -330,16 +342,15 @@ function createProgram(args, initialInputs) {
         compilerOptions.extendedDiagnostics && host.trace?.(message)
     }
 
-    function getDirectoryWatcherForPath(path) {
-        for (const [directory, callbacks] of directoryWatchers.entries()) {
-            if (isAncestorDirectory(directory, path)) {
-                debuglog(`found a directory watcher for ${path} ${directory}`);
-                return (filePath) => {
-                    for (const callback of callbacks) {
-                        callback(filePath)
-                    }
-                };
-            }
+    function getDirectoryWatcherForPath(fileName) {
```

Comment:
Seems like the only time we call this we just invoke the return method right away... should we just do `invokeDirectoryWatcherForPath`?

Reply:
Right now we're doing 3 things though: collect the callbacks, create a wrapper function, invoke the wrapper function. Why not just do one thing: invoke the callbacks?

Reply:
Yes. I want to hide the fact that there could be more than one callback for directory watchers. Invoking is the job for the invalidate function and keep it consistent with fileWatchers.

Reply:
Yeah that is clearer to me, and while only a minor difference I find it easier to ready 👍 🤷